notebooks/tutorials/overland_flow/overland_flow_erosion/ol_flow_erosion_components.ipynb

```diff
@@ -0,0 +1,489 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a href=\"http://landlab.github.io\"><img style=\"float: left\" src=\"../../../landlab_header.png\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Components for modeling overland flow erosion \n",
+    "\n",
+    "*(G.E. Tucker, July 2021)*\n",
+    "\n",
+    "There are two related components that calculate erosion resulting from surface-water flow, a.k.a. overland flow: `DepthSlopeProductErosion` and `DetachmentLtdErosion`. They were originally created by Jordan Adams to work with the `OverlandFlow` component, which solves for water depth across the terrain. They are similar to the `StreamPowerEroder` and `FastscapeEroder` components in that they calculate erosion resulting from water flow across a topographic surface, but whereas these components require a flow-routing algorithm to create a list of node \"receivers\", the `DepthSlopeProductErosion` and `DetachmentLtdErosion` components only require a user-identified slope field together with an at-node depth or discharge field (respectively)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## `DepthSlopeProductErosion`\n",
+    "\n",
+    "This component represents the rate of erosion, $E$, by surface water flow as:\n",
+    "\n",
+    "$$E = k_e (\\tau^a - \\tau_c^a)$$\n",
+    "\n",
+    "where $k_e$ is an erodibility coefficient (with dimensions of velocity per stress$^a$), $\\tau$ is bed shear stress, $\\tau_c$ is a minimum bed shear stress for any erosion to occur, and $a$ is a parameter that is commonly treated as unity.\n",
+    "\n",
+    "For steady, uniform flow, \n",
+    "\n",
+    "$$\\tau = \\rho g H S$$,\n",
+    "\n",
+    "with $\\rho$ being fluid density, $g$ gravitational acceleration, $H$ local water depth, and $S$ the (postive-downhill) slope gradient (an approximation of the sine of the slope angle).\n",
+    "\n",
+    "The component uses a user-supplied slope field (at nodes) together with the water-depth field `surface_water__depth` to calculate $\\tau$, and then the above equation to calculate $E$. The component will then modify the `topographic__elevation` field accordingly. If the user wishes to apply material uplift relative to baselevel, an `uplift_rate` parameter can be passed on initialization.\n",
+    "\n",
+    "Here are the header and constructor docstrings:"
```

Comment:
needs a bit of explanation - what's the header, what's the constructor?

Reply:
explanation added